### DISCUSSION TITLE
Ascii Terminal HW

### DISCUSSION THREAD

**Discussion by olaf-h** [thehighnibble/thehighnibble.github.io#122] [IMSAI8080esp replica]
Hi Guys I've just made Geoff Grahams marvelous ASCII terminal emulator. The HW is working fine, but I am not able to connect it to the 8080 over the serial port. How do I set up one of the serial ports for usage with an external ASCII terminal?

**Comment by olaf-h** [comment]
A short update after some trying: I receive output, but it is not readable. I will experiment with baud rates and other settings, but any help is still appreciated

**Reply by nealzip** [reply]
@olaf-h I'm still waiting on my order for a IMSAI8080. I know serial protocols very well but without a IMSAI or Geoff's board  I can't help on that side. 

One suggestion I can make is on Geoff''s board you should be able to connect pin's 2 and 3 on the DB9  and see your typing coming back to you. That is a trick I used long ago to test serial ports. At least that will tell you that the Terminal Emulator is working as expected. 

But it sounds like you are on the right track of changing Baudrate and/or parity. 

Only other item I can think of is Serial port signal level 5v vs +/-15v. But without knowing/having the two pieces of hardware it might be prudent to hold off getting that deep yet.

Hope some of this info is helpful. I try to check back with you to see how troubleshooting is going.
//Neal

**Reply by olaf-h** [reply]
Thanks for your suggestion. Echo is working on the serial port. But I figured out that it might be a TTL / RS-232 setup issue. The ASCII board should be able to handle both inputs. I will check with the imsai what is used. Maybe that is the solution :-)

**Reply by nealzip** [reply]
The Ttl/RS-232 to is the same iisue as the 5v / +-15v issue. Here is a level shifter i have used to make the conversion.

https://www.sparkfun.com/products/449

Good luck.
